frontend/src/pages/admin/Template/sections/Catalogues.vue

```diff
@@ -0,0 +1,186 @@
+import LockSetting from '../components/LockSetting.vue'
+
+export default {
+    name: 'TemplateCatalogueEditor',
+    components: {
+        FormRow,
+        FormHeading,
+        ChangeIndicator,
+        LockSetting,
+        PlusSmIcon,
+        XIcon,
+        UndoIcon
+    },
+    props: {
+        editTemplate: {
+            type: Boolean,
+            default: false
+        },
+        modelValue: {
+            type: Object,
+            required: true
+        },
+        readOnly: {
+            type: Boolean,
+            default: false
+        },
+        project: {
+            type: Object,
+            required: false,
+            default: null
+        }
+    },
+    emits: ['update:modelValue'],
+    data () {
+        return {
+            defaultCatalogue: 'https://catalogue.nodered.org/catalogue.json',
+            showAddRow: false,
+            input: {
+                url: '',
+                error: ''
+            },
+            urls: []
+        }
+    },
+    computed: {
+        editable: {
+            get () {
+                return this.modelValue
+            },
+            set (localValue) {
+                this.$emit('update:modelValue', localValue)
+            }
+        },
+        projectLauncherCompatible () {
+            if (this.editTemplate) {
+                // When editing template we don't have a project
+                return true
+            }
+
+            const launcherVersion = this.project?.meta?.versions?.launcher
+            if (!launcherVersion) {
+                // We won't have this for a suspended project - so err on the side
+                // of permissive
+                return true
+            }
+            // TODO needs to semver >= 1.12.0
+            return SemVer.satisfies(launcherVersion, '>=1.11.0')
+        },
+        addDefaultEnabled () {
+            if (this.readOnly) {
+                return true
+            } else {
+                return this.urls.includes(this.defaultCatalogue)
+            }
+        },
+        defaultEnabled () {
+            // whether or not this Template has the default catalogue enabled
+            return this.urls.includes(this.defaultCatalogue)
+        },
+        thirdPartyUrls () {
+            // whether or not this Template has any third party catalogues enabled
+            return this.urls.filter(url => url !== this.defaultCatalogue)
+        }
+    },
+    watch: {
+        'editable.settings.palette_catalogue': {
+            deep: true,
+            handler (v) {
+                this.urls = v
+            }
+        }
+    },
+    mounted () {
+        // deep copy
+        this.urls = JSON.parse(JSON.stringify(this.editable.settings.palette_catalogue))
+    },
+    methods: {
+        addURL () {
+            const newURL = this.input.url.trim()
+            if (newURL) {
+                try {
+                    // eslint-disable-next-line
+                    const u = new URL(newURL)
```

Comment:
This can just be `new URL(newURL)` believe

Reply:
No, because lint complains

Reply:
But you've disabled the linter ;-)